rest/server_context.go

```diff
@@ -480,6 +485,49 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(config *DbConfig, useExisti
 		if validateErr := replicationConfig.ValidateReplication(true); validateErr != nil {
 			return nil, validateErr
 		}
+
+		// Only support SGR1 checkpoint upgrade if the replication is unidirectional.
+		if replicationConfig.Direction == db.ActiveReplicatorTypePush || replicationConfig.Direction == db.ActiveReplicatorTypePull {
+			// SGR1 checkpoint fallback:
+			// If we match based on replication ID, we'll generate the SGR1 checkpoint ID and use it as a fallback in the event that no
+			// SGR2 checkpoints can be found. This needs to be persisted cluster-wide so that an assigned node that is not running SGR1 replications can still utilise the existing checkpoint.
+			// SGR1 replications were unidirectional only, and so we can't match replication IDs for SGR1 checkpoints in the event of a pushAndPull SGR2 replication.
+			for _, sgr1ReplicationConfig := range sc.config.Replications {
+				if sgr1ReplicationConfig.ReplicationId != replicationID {
+					continue
+				}
+				base.Debugf(base.KeyReplicate, "Matched replication IDs for SGR1 checkpoint fallback: %v %v", replicationConfig, sgr1ReplicationConfig)
+
+				// TODO: We don't ensure SGR1 and SGR2 params match, it might be possible for users to change replication filters when moving to SGR2 and invalidate the SGR1 checkpoint.
+
+				// don't have running replications available before startup, so regenerate the params in this case for us to generate a checkpoint ID.
+				params, _, localdb, err := validateReplicateV1Parameters(*sgr1ReplicationConfig, true, *sc.config.AdminInterface)
+				if err != nil {
+					base.Warnf("Couldn't build SGR1 replication parameters for replication config: %v", err)
+					break
+				}
+				if !localdb {
```

Comment:
Shouldn't we also be checking that the localdb being targeted (either source or target, depending on direction) matches this database, and isn't targeting a different db?  I don't think we enforce sgr2 replicationID uniqueness across databases.